### ISSUE TITLE
Advice on which yolo version use for difficult dataset

### ISSUE THREAD

**Issue by drapado** [AlexeyAB/darknet#2715]
Hi, I'm using yolo in my custom dataset (1000x1000 image resolution).

My dataset has object from the same class that with very different sizes, from 7x7 pixels (from a 1000x1000 image) to 200x200 pixels.

I've tried both yolo-v3-5l and yolo-v3-spp but even when I get very good mAP values for the train set, I'm getting around 20% on the validation set (both sets come from the same distribution). This sounds clearly like overfitting, however I never reach more than 20% mAP in the valid set during the whole training process.

Which version of yolo do you suggest me to use? Also, any other advice?

Thank you in advance!

**Comment by AlexeyAB** [comment]
>  get very good mAP values for the train set, I'm getting around 20% on the validation set (both sets come from the same distribution).

* Did you get Validation dataset by random picking up images from initial dataset? Or did you get last X % of images from initial dataset?

* What mAP did you get on Training dataset?

* In you case you can try to use `yolo-v3-5l.cfg` with `width=1024 height=1024` if you have enough GPU-RAM and you are satisfied with the speed of detection.
   If not, then try to use `yolov3-tiny-3l.cfg` with `width=1024 height=1024`

* Try to train with default anchors if you calculated custom anchors.

* Also can you show screenshot of cloud of point `cloud.png`
`./darknet detector calc_anchors data/obj.data -num_of_clusters 9 -width 1024 -height 1024 -show`

**Comment by drapado** [comment]
@AlexeyAB thanks for answering,

1. Yes, I random picked them
2. Around 80%
3. I'll try with tiny yolo since I'm using a rtx 2060
4. I'll try, but why will it be better with the default anchors?
![DeepinScreenshot_clusters_20190324133007](https://user-images.githubusercontent.com/45577687/54879424-21484e80-4e39-11e9-8edf-91e5ccd68131.png)

**Comment by AlexeyAB** [comment]
@drapado

It seems most of your objects are very small.

What calculated anchors do you get?

**Comment by drapado** [comment]
@AlexeyAB 
anchors =  12, 11,  25, 21,  34, 49,  58, 31,  66, 66, 125, 72,  96,121, 150,156, 211,216

**Comment by drapado** [comment]
@AlexeyAB I'm currently training yolo-5l with width=736 and height=736 because of ram, but I was able to train a yolo-tiny-3l with 1024x1024. The performance improved, although I still got 42% mAP on validation set (89% on train set) which was more or less constant from iteration 2000 to iteration 10000 (40-43% of mAP). I just have 4 classes, neither the loss nor the mAP seemed to change since iteration 2000-3000 (loss around 3.x).

Any advice? Should I train during more iterations?

**Comment by AlexeyAB** [comment]
@drapado 

What model gives you higher mAP ` yolo-tiny-3l with 1024x1024` or `yolo-5l with width=736 and height=736` ?

Also can you show full output of ./darknet detector map ... command? How many TP/FP/FN...?

> Any advice? Should I train during more iterations?

You can try to reduce learning_rate 10x and continue training.

**Comment by drapado** [comment]
I´m still waiting for the training process of yolov3-5l 736x736 to finish because it takes quite a while, but with almost 3000 iterations it seems to perform slightly worse than yolo-tiny-3l 1024x1024
